### DISCUSSION TITLE
Example for set size

### DISCUSSION THREAD

**Discussion by mvolar** [PaulleDemon/tkVideoPlayer#19] [Q&A] [answered]
Hello,

I love your package, however I am having some trouble implementing the .set_size() method in a working Tkinter mainloop, so if you could be so kind as to provide a minimal example of manual video size setting?

The problem is when I embed the video in the Tkinter window it only shows a corner of the video in a very small box. I manually set the size using this code:
```
vid_player = TkinterVideo( master=root,keep_aspect=False,scaled=True)
vid_player.set_size(size=(600,600))
```

**Comment by PaulleDemon** [comment, marked as answer]
Hello @mvolar, `set_size` sets the video size, not the window or the frame.

So what you probably want is `root.geometry("500x500)`  if you want to specify the position then you can use `root.geometry('wxh+x+y')`

```
import tkinter as tk
from tkVideoPlayer import TkinterVideo

root = tk.Tk()

root.geometry('500x500') # < ---------- This is what you need

tkvideo = TkinterVideo(scaled=False, master=root)
tkvideo.load(r"test.mp4")
tkvideo.pack(expand=True, fill="both")
tkvideo.set_size((1200, 600)) # sets the video  size
tkvideo.play() # play the video

root.mainloop()
```

If  this solves your problem, make sure to mark this as the answer; if you have any further questions feel free to ask

**Reply by mvolar** [reply]
The video was abnormally sized, only when using both the root geometry and set_size() method have I gotten it to work. Many thanks.

I just have one simple question regarding the frame generated event from https://github.com/PaulleDemon/tkVideoPlayer/issues/6. How would you write a function which returns the number of the frame upon as an integer? I tried using framecounters but it only breaks everthing (including the kernel) since it has a lot of function calls.

**Reply by PaulleDemon** [reply]
@mvolar `tkvideo.video_info()` should provide you with enough information to determine the number of frames. You might have to do some math to get it, if you are finding it hard let me know.

**Reply by mvolar** [reply]
I managed to use the video info together with the time passes from pressing the buttons in order to get the time/rough estimate of frames passes (I afterwards perform some analysis on the stopped frame). Trouble arises because of https://github.com/PaulleDemon/tkVideoPlayer/issues/15 because the framerate*time!=frames_generated. The performance framerate being to fast/slow isn't an issue currently but I would really like to get the approximate frame play/pause button was pressed.

Using it like this doesn't break anything but a video isn't being played, it seems like the video player binding for `<<FrameGenerated>>` virtual event cannot perform 2 functions, so it either loads and displays the frame, or reports the event. The solution would probably be to wirte a function which both displays the frame AND reports the event/dumps it in a variable .If you could provide any help it would be much appreciated.

Using the sample player defined by you i just defined a new function:

```
def get_every_25_frame(event):
    global framecount
    framecount+=1
    if framecount%25==0:
        print (framecount)
        
```
and binded it to the vid_player

```
vid_player.bind("<<Duration>>", update_duration)
vid_player.bind("<<SecondChanged>>", update_scale)
vid_player.bind("<<Ended>>", video_ended )
vid_player.bind("<<FrameGenerated>>", get_every_25_frame ) #shows framecount doesn't show video
```

however:

```
vid_player.bind("<<Duration>>", update_duration)
vid_player.bind("<<SecondChanged>>", update_scale)
vid_player.bind("<<Ended>>", video_ended )
vid_player.bind("<<FrameGenerated>>", get_every_25_frame ) #shows video doenst show frame_count
vid_player.bind("<<FrameGenerated>>", vid_player._display_frame)

```

**Reply by PaulleDemon** [reply]
Hey @mvolar , sorry for the late reply. The easiest way to get a frame number is by using `vid_player._frame_number`.

eg:
```py
import tkinter as tk
from tkVideoPlayer import TkinterVideo

def get_frame_count():
    print("Video info: ", tkvideo._frame_number)

root = tk.Tk()

tkvideo = TkinterVideo(scaled=False, master=root)
tkvideo.load(r"test.mp4")
tkvideo.pack(expand=True, fill="both")
tkvideo.set_size((1200, 600)) # sets the frame size
tkvideo.play() # play the video

btn = tk.Button(root, text="get frame count", command=get_frame_count)
btn.pack()

root.mainloop()
```
